armory/art_experimental/attacks/carla_adversarial_texture.py

```diff
@@ -48,24 +42,40 @@ def generate(self, x, y, y_patch_metadata=None, **kwargs):
         x_min = int(np.min(gs_coords[:, 1]))
         y_min = int(np.min(gs_coords[:, 0]))
 
-        attack = AdversarialTexturePyTorch(
-            self.estimator,
-            patch_height=patch_height,
-            patch_width=patch_width,
-            x_min=x_min,
-            y_min=y_min,
-            **self.attack_kwargs
+        self.patch_height = patch_height
+        self.patch_width = patch_width
+        self.x_min = x_min
+        self.y_min = y_min
+
+        # reinitialize patch
+        self.patch_shape = (patch_height, patch_width, 3)
+        mean_value = (
+            self.estimator.clip_values[1] - self.estimator.clip_values[0]
+        ) / 2.0 + self.estimator.clip_values[0]
+        self._initial_value = np.ones(self.patch_shape) * mean_value
+        self._patch = torch.tensor(
+            self._initial_value, requires_grad=True, device=self.estimator.device
         )
 
         # this masked to embed patch into the background in the event of occlusion
         foreground = y_patch_metadata[0]["masks"]
         foreground = np.array([foreground])
 
+        # create patch points indicating locations of the four corners of the patch in each frame
+        patch_points = []
```

Comment:
It looks like the line `patch_points = []` can be deleted since `patch_points` is then subsequently reset in the conditional block that follows

As a side note: do we know of a case where `gs_coords.ndim != 2`? Is the point of this conditional logic that `gs_coords` could be of shape `(num_frames, 4, 2)` and that in this case the `else` block is reached?

Reply:
Yes, `patch_points = []` can be removed.

Yes, Eval 5 videos will introduce some camera motion so the `gs_coords` will have shape `(num_frames, 4, 2)`